src/many-ledger/src/storage.rs

```diff
@@ -469,30 +481,41 @@ impl LedgerStorage {
         use rocksdb::{Direction, IteratorMode, ReadOptions};
 
         let mut options = ReadOptions::default();
-        let mut bound = TRANSACTIONS_ROOT.to_vec();
+        options.set_iterate_lower_bound(MULTISIG_TRANSACTIONS_ROOT);
+        let mut bound = MULTISIG_TRANSACTIONS_ROOT.to_vec();
         bound[MULTISIG_TRANSACTIONS_ROOT.len() - 1] += 1;
-        options.set_iterate_upper_bound(bound);
 
-        let it = self.persistent_store.iter_opt(
-            IteratorMode::From(TRANSACTIONS_ROOT, Direction::Forward),
-            options,
-        );
+        let it = self
+            .persistent_store
+            .iter_opt(IteratorMode::From(&bound, Direction::Reverse), options);
 
-        let mut keys_to_delete: Vec<Vec<u8>> = Vec::new();
+        let mut batch = vec![];
 
         for (k, v) in it {
             let v = Tree::decode(k.to_vec(), v.as_ref());
 
-            let storage: MultisigTransactionStorage = minicbor::decode(v.value())
+            let mut storage: MultisigTransactionStorage = minicbor::decode(v.value())
                 .map_err(|e| ManyError::deserialization_error(e.to_string()))?;
+            let now = self.now();
 
-            if storage.info.timeout.0 > self.current_time.unwrap_or_else(SystemTime::now) {
-                keys_to_delete.push(k.to_vec());
+            if storage.info.timeout.0 > now {
+                if !storage.disabled {
+                    storage.disabled = true;
+                    storage.info.state = MultisigTransactionState::Expired;
+
+                    if let Ok(v) = minicbor::to_vec(storage) {
+                        batch.push((k.to_vec(), Op::Put(v)));
+                    }
+                }
+            } else if let Ok(d) = now.duration_since(storage.creation) {
```

Comment:
Since the DB is ordered, at this point we don't need to continue since we know that the rest is all timed out anyway.

Reply:
I assume you want me to add a comment?  ^_^

Reply:
If you want; it was mostly for my comprehension.